tests/Controller/ExportControllerTest.php

```diff
@@ -228,4 +214,189 @@ public function testCorrectTranslatedProductIsExported(): void
+
+        $this->getContainer()->get('sales_channel.repository')->update([
+            [
+                'id' => $salesChannelContext->getSalesChannel()->getId(),
+                'domains' => [
+                    [
+                        'currencyId' => $currencies->first()->getId(),
+                        'snippetSetId' =>
+                            $salesChannelContext->getSalesChannel()->getDomains()->first()->getSnippetSetId(),
+                        'url' => 'http://cool-url.com/german'
+                    ]
+                ]
+            ],
+        ], Context::createDefaultContext());
+
+        $this->enableFindologicPlugin(
+            $this->getContainer(),
+            self::VALID_SHOPKEY,
+            $salesChannelContext
+        );
+
+        $product = $this->createVisibleTestProduct([
+            'visibilities' => [
+                [
+                    'id' => Uuid::randomHex(),
+                    'salesChannelId' => $salesChannelContext->getSalesChannel()->getId(),
+                    'visibility' => 20
+                ]
+            ],
+            'seoUrls' => []
+        ]);
+
+        // Explicitly remove all SEO URLs, since they were automatically created when using DAL for product creation.
+        Kernel::getConnection()->executeUpdate('DELETE FROM seo_url');
+
+        $response = $this->sendExportRequest();
+
+        $this->assertSame(200, $response->getStatusCode());
+        $this->assertSame('text/xml; charset=UTF-8', $response->headers->get('content-type'));
+        $parsedResponse = new SimpleXMLElement($response->getContent());
+
+        $this->assertSame(1, (int)$parsedResponse->items->attributes()->count);
+        $this->assertSame($product->getId(), $parsedResponse->items->item->attributes()->id->__toString());
+        $this->assertSame(
+            'http://localhost/german/detail/' . $product->getId(),
+            $parsedResponse->items->item->urls->url->__toString()
+        );
+    }
+
+    protected function sendExportRequest(array $overrides = []): Response
+    {
+        $defaults = [
+            'shopkey' => self::VALID_SHOPKEY
+        ];
+
+        $params = array_merge($defaults, $overrides);
+        $client = $this->getTestClient($this->salesChannelContext);
+        $client->request('GET', '/findologic?' . http_build_query($params));
+
+        return $client->getResponse();
+    }
+
+    protected function parsePluginVersion(): string
+    {
+        $composerJsonContents = file_get_contents(__DIR__ . '/../../composer.json');
+        $parsed = json_decode($composerJsonContents, true);
+
+        // For release candidates, Shopware will internally format the version different, from how
+        // it is set in the composer.json.
+        return str_replace('rc.', 'RC', ltrim($parsed['version'], 'v'));
+    }
+
+    /**
+     * Unlike SalesChannelHelper::buildSalesChannelContext, which modifies the default sales channel, this
```

Comment:
Why not rename `buildSalesChannelContext` to `modifySalesChannelContext`?

Reply:
That comment is not really accurate. It should be

... which **by default** modifies the default sales channel, this ...

We actually use it in this method to create an entirely new one, as `buildSalesChannelContext` uses `upsert`, which can either modify existing data, or if it doesn't exist, creates that data.

Updated the docblock.